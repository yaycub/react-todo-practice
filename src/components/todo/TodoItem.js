import React from 'react';
import PropTypes from 'prop-types';

const TodoItem = ({ name, completed, handleDelete, styles }) => {

  return (
    <>
      <h3 className={completed ? styles?.completed : styles?.incomplete}>Todo: {name}</h3>
      <button onClick={() => handleDelete(name)}>Delete Todo</button>
    </>
  );
};

TodoItem.propTypes = {
  name: PropTypes.string.isRequired,
  completed: PropTypes.bool.isRequired,
  handleDelete: PropTypes.func.isRequired,
  index: PropTypes.number.isRequired,
  styles: PropTypes.shape({
    completed: PropTypes.string,
    incomplete: PropTypes.string
  }).isRequired
};

export default TodoItem;
