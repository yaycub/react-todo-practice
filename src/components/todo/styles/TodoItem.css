.completed{
  text-decoration: line-through;
}

.incomplete{
  text-decoration: solid;
}