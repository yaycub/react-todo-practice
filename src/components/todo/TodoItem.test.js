import React from 'react';
import { shallow } from 'enzyme';
import TodoItem from './TodoItem';

describe('TodoItem Component', () => {
  it('should render a TodoItem', () => {
    const wrapper = shallow(<TodoItem />);
    expect(wrapper).toMatchSnapshot();
  });
});
