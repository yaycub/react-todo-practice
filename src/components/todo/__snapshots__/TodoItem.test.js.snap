// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoItem Component should render a TodoItem 1`] = `
<Fragment>
  <h3>
    Todo: 
  </h3>
  <button
    onClick={[Function]}
  >
    Delete Todo
  </button>
</Fragment>
`;
